src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -294,6 +304,48 @@ class LinalgBackendEigen : public LinalgBackendBase
 		result_eig = alpha * a_eig + beta * b_eig;
 	}
 
+	/** Eigen3 convolve/cross-correlation method. */
+	template <typename T>
+	void convolve_impl(SGMatrix<T>& X, SGMatrix<T>& W, SGMatrix<T>& Y,
+		bool flip=false, index_t stride_x=1, index_t stride_y=1)
+	{
+		int32_t width = X.num_cols;
+		int32_t height = X.num_rows;
+
+		int32_t kx = W.num_cols;
+		int32_t ky = W.num_rows;
+
+		int32_t rx = (kx-1)/2;
+		int32_t ry = (ky-1)/2;
+
+		for (int32_t x=0; x<width; x+=stride_x)
+		{
+			int32_t xout = x/stride_x;
+
+			for (int32_t y=0; y<height; y+=stride_y)
+			{
+				int32_t yout = y/stride_y;
+
+				T sum = 0;
+				for (int32_t x1=x-rx; x1<=x+rx; x1++)
+				{
+					int32_t wx = flip ? x1-x+rx : rx-x1+x;
+					for (int32_t y1=y-ry; y1<=y+ry; y1++)
+					{
+						if (x1>=0 && y1>=0 && x1<width && y1<height)
```

Comment:
grep for convlution here:
http://eigen.tuxfamily.org/index.php?title=3.3

We can take code from there, this one should not be in, it is slow and unreadable. I know it is not your fault, but if we have unit tests it should be easy to update the code.....